### ISSUE TITLE
[Suggestion] Fluid Pump

### ISSUE THREAD

**Issue by zhs2** [CoFH/Feedback#277]
To my knowledge, there is nothing in the Thermal suite that transfers world fluids to liquid inventories. If this existed in a Thermal mod, there would be no need to include another mod that supplies a pump in a modpack, such as Buildcraft or Mekanism.

**Comment by Tora-B** [comment]
The Terrain Smasher still works on fluids, and will output them to a fluiduct, but it would be nice to have a computationally efficient, large-scale pump for collecting fluids. My guess is that the challenges to designing a good pump are the algorithm for picking a block, and minimizing the number or impact of the block updates caused by changing the world.

Thermal Expansion already has the Aqueous Accumulator for producing water in a server-friendly way, so it might be more useful if the proposed pump was designed to prevent water from generating new source blocks.

**Comment by zhs2** [comment]
Or just ignored water, much as it would be nice to have a pump that would destroy an underground lake at will - I do believe that's what strategic placement of TE Sponges might be for, however. Didn't know the Terrain Smasher could collect fluids, though.

**Comment by KingLemming** [comment]
It's not that a pump (or quarry) is hard to do, it's that I've actively tried to not obsolete BuildCraft, despite what people may say.

It's been a fairly strong "maybe" and it's something that we'll likely tackle in the future. But I can't give anything even remotely approaching an ETA.

**Comment by zhs2** [comment]
If I may be fair, I think Buildcraft itself is obsolete. There are a lot of things Buildcraft does in a "not-so-convenient" kind of way that makes me feel, personally, as if there are other mods out there that fulfill the same purpose as Buildcraft, except more intelligently designed. IC2 and Mekanism (though, IC2 doesn't have any sort of transfer pipes) have long since outshone Buildcraft's capabilities, and so has Thermal Expansion...

Thanks for the consideration.

**Comment by BoboSims** [comment]
If a 'Thermal' alternative to the BC pump were to exist - regardless of efficiency -, I would probably find new solutions for some of my piping, and get rid of BC altogether. Because besides said pump, it offers me very little that I cannot do without.

BC has long been the way to go for many tech/energy solutions. A long time ago, *BC* introduced energy into the game, but RF now rules the waves. *BC* got excellent piping at a reasonable resource cost, but recently their kinesis pipes have begun to refuse to deliver RF to most other mods' blocks, even if said blocks *do* accept RF. So if BC is to be called obsolete, it must be for the right reason: they are going the way of the dinosaur, because **they** failed to evolve with current technology. Whether you design a pump or not, BC's fate is **not your responsibility**!!!

**Comment by Frontrider** [comment]
That is the one and only block to have from that mod.

The lava pump is kida solved in a server friendly way by extra utilities, so i would say something with a smaller scale to pump lakes.

**Comment by BoboSims** [comment]
@Frontrider: Agreed! Since my previous post, I have already completely abandoned BC pipes, so only the pump and the quarry remain. And I could do without the quarry in a heartbeat.

For Items/Fluids, it's now all TD conduits, and for Energy/Redstone it's all MFR cables. These are so far the most versatile solutions around.

**Comment by Frontrider** [comment]
Enderquarry is your solution for that problem. Extra utilities contains enough stuff to be usefull, whitout overlapping.

**Comment by BoboSims** [comment]
Extra Utilities and Ender IO also heavily rely on Ender Pearls... being a Peaceful-player (an inventor, not a mob-hunter) that makes much of their stuff hard for me to build/use.

But as I said, I can live without the quarry. When I'm mining by hand, I can at least use silk touch when needed.

**Comment by Frontrider** [comment]
I'm a peacefull player too. Use a magnum torch to prevent natural mob spawning, then cursed earth to spawn things.
Extra utilites contains a block called peacefull table. Give it a diamond sword, and it will give you drops.
Also there are ender lilys to grow ender pearls.

**Comment by BoboSims** [comment]
Yeah, well, there are problems...
The magnum torch and cursed earth... don't these require some high-end mob-dropped items as well? I would love to *find* some Ender lilies first... even *one* would help. But so far, no joy. As long as I don't have even one of them, I'm not going anywhere! :D

**Comment by Frontrider** [comment]
Peacefull table: some emeralds.
the magnum torch does require some things, but the cursed earth technically not.

i don't think that this belongs here, so we should stop.

**Comment by zhs2** [comment]
Not to be the dick of my own comment thread, but "just use this other mod's block!" does not make for a helpful post. Granted, there will be preferences, and there will be some pumps that are better than others (I know I absolutely love Mekanism's because it does what it is supposed to and I don't have to hook an energy block to it), but the premise of this thread (at least, the premise I intended) was feature completeness. (Plus, TE machines are generally sexy. I'm still getting over a huge boner for the Phytogenic Insolator and want to hug the guy who invented it. No homo.)

**Comment by BoboSims** [comment]
Suggestions is all they were, if you ask me. Suggestions that *might* - at least *temporarily* - fill some of our requirements. I don't think there's anything ambiguous or unclear about your intentions. It's very clear what you're looking for, and I want the same, mostly. I don't think there's anything wrong with suggestions either, though.

And yes, those Thermal *** blocks are hot! And not only in looks, but in the way they work and are configured as well. Their interfaces work for me!

**Comment by Frontrider** [comment]
@zhs2 Thats not ment to be the thing :D.

Completly agreed with you. I love every thermal expansion machine, they look/function really well. The pytogenic insolator is a really nice machine. Extra point for being 100% functional with pam's harvestcraft plants.

I just said whats possible to do at this very point.

**Comment by BoboSims** [comment]
I *do* hope, though, that when we *do* get a pump, a device similar to the Flood Gate (the inverse of the Pump) will also become available. Or maybe the pump-style device could be easily reconfigured to do the reverse. One way pumping fluids from the world into pipes and tanks, and the other way pumping fluids from tanks and pumps into the world. :smile:

**Comment by KingLemming** [comment]
It's a 2-way, yes. That's why it's taking a while. That and lack of time...

**Comment by BoboSims** [comment]
Take as long as you need, @KingLemming. The creative process can't be rushed without adversely affecting the quality.

**Comment by zhs2** [comment]
Two-way? Wow. I always thought the idea, like Mekanism's Fluidic Plenisher, to be kind of silly: combine with Aqueous Accumulator at the top of the world for maximum lulz.

**Comment by stclaus4** [comment]
if you have enderIO installed there is a new add-on for it, this adds a block that add a revers pump type action therefore newer mods HAVE made buildcraft officially obsolete. sorry. eh?
